### PR TITLE
Opentracing Support

### PR DESCRIPTION
**Pull request by sdwr98**
Addresses #258 - bringing over the ContextPropagator improvements and adding a default OpenTracing implementation of context propagation

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      HashMapTextMap contextTextMap = new HashMapTextMap(contextAsMap);
+      setCurrentOpenTracingSpanContext(
+          currentTracer.extract(Format.Builtin.TEXT_MAP, contextTextMap));
+    }
+  }
+
+  @Override
+  public void setUp() {
+    Tracer openTracingTracer = GlobalTracer.get();
+    Tracer.SpanBuilder builder = openTracingTracer.buildSpan("cadence.workflow");
```
Comment (mfateev) [inline temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java:117]: I don't think it is going to work. The problem is that during replay a new span is going to be created. 
Unfortunately, there is no way to create a span passing deterministically generated ID to it. 

Another option would be using [SideEffect](https://github.com/temporalio/sdk-java/blob/4bb23c3a660de6a2630901d0dc6c049d9742ebe1/temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java#L920) to store the span, but that implies additional MarkerEvent for every workflow.

File: temporal-sdk/src/main/java/io/temporal/common/context/ContextPropagator.java [hunk 2]
```diff
@@ -126,4 +130,27 @@
 
   /** Sets the current context */
   void setCurrentContext(Object context);
+
```
Comment (mfateev) [inline temporal-sdk/src/main/java/io/temporal/common/context/ContextPropagator.java:133]: Looking at this and the implementation I'm getting more inclined to just using interceptors for this integration and getting rid of the context propagator interface completely as it has very limited use.

Comment (sdwr98) [reply temporal-sdk/src/main/java/io/temporal/common/context/ContextPropagator.java:133]: I’d be happy to look into that - are there any docs or examples of interceptors anywhere? I didn’t see a way of propagating information from worker/client to the server and back using interceptors.

Comment (mfateev) [reply temporal-sdk/src/main/java/io/temporal/common/context/ContextPropagator.java:133]: I think you are right that there is something missing in the interceptors to support this use case. Let me think about this.

File: temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java (hunk 1 above)
Comment (sdwr98) [reply temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java:117]: Does it really matter if we start multiple spans if we only finish one?  I don't think spans get reported until you finish() them.

File: temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    }
+  }
+
+  @Override
+  public String getName() {
+    return "OpenTracing";
+  }
+
+  @Override
+  public Map<String, Payload> serializeContext(Object context) {
```
[the remark this answers is not in the document]
Comment (sdwr98) [reply temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java:66]: That's a good point - I will take a look at the go propagator and work on ensuring cross-platform.

**Comment by mfateev** [comment]
The newly added [WorkerInterceptor](https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/common/interceptors/WorkerInterceptor.java) should be enough to implement open tracing support.

I'm going to close this PR. I would love to review a contribution that relies on the interceptor.

**Pull request closed without merging**